plugin.js

```diff
@@ -42,9 +38,10 @@ function plugin (fn, options = {}) {
 
   if (!options.name) {
     autoName = true
-    options.name = pluginName + '-auto-' + count++
+    options.name = checkName(fn) + '-auto-' + count++
   }
 
+  fn[Symbol.for('skip-override')] = !options.encapsulate
```

Comment:
```suggestion
  fn[Symbol.for('skip-override')] = options?.encapsulate ?? true
```

Reply:
A few lines above we make sure `options` is a non-null object

Reply:
I would still rather not rely on a difficult to interpret coercive `!`.

Reply:
If we check before that options is a POJO then we can avoid the question mark. But regarding the coercion with an exclamation mark i agree with jsumners

```suggestion
  fn[Symbol.for('skip-override')] = options.encapsulate ?? true 
```